lib/hal/xbox.c

```diff
@@ -68,7 +95,17 @@ void XLaunchXBEEx(const char *xbePath, const void *launchData)
 
     if (!xbePath) {
         launchDataPage->Header.dwLaunchDataType = LDT_LAUNCH_DASHBOARD;
-    } else {
+
+        // Redirect to xonlinedash.xbe as needed.
+        if (launchData && *(int*)launchData >= LDT_LAUNCH_DASHBOARD_REASON_NETWORK) {
```

Comment:
In `XGetLaunchInfo` we appear to use `*(unsigned long*)` for the `launchData` type field. From what I can tell, casting to `PLAUNCH_DATA_HEADER` and then accessing `dwLaunchDataType` would be even cleaner (`DWORD`).

Reply:
launchData is not a `PLAUNCH_DATA_HEADER`, it's the arbitrary user defined data after the header and padding. I cast to a DWORD, but this isn't a struct that's defined in the nxdk at all (but maybe it should be for this one magic use case?)

Reply:
Actually, this will break any homebrew titles that may use `DWORD` size variable in the first offset of launchData. Reason? launchData is copy directly to launchDataPage's LaunchData offset. Therefore doesn't set any to launchDataPage's Header offset other than szLaunchPath member variable from the first parameter of XLaunchXBEEx function.

The parameter of launchData from XLaunchXBEEx function is intended to pass on directly without read from it. So, we shouldn't do this way.

Reply:
This will only affect titles that use `XLaunchXBEEx` specifically to launch the dashboard by failing to pass an `xbePath`. At that point, I believe the structure of the launch data is (semi) well defined, since the target being launched sets the expectations for the content. This is why there's a `dwLaunchDataType` enumeration that calls out the fact that it's of "LAUNCH_DASHBOARD" type.